server/punishments.ts

```diff
@@ -363,6 +369,36 @@ export const Punishments = new class {
 		return FS(SHAREDIPS_FILE).write(buf);
 	}
 
+	/**
+	 * sharedips.tsv is in the format:
+	 * IP, type (in this case always SHARED), note
+	 */
+	async loadSharedIpBlacklist() {
+		const data = await FS(SHAREDIPS_BLACKLIST_FILE).readIfExists();
+		if (!data) return;
+		for (const row of data.split("\n")) {
+			if (!row || row === '\r') continue;
+			const [ip, reason] = row.trim().split("\t");
+			if (!ip.includes('.')) continue;
```

Comment:
```suggestion
			if (!IP_REGEX.test(ip)) continue;
```
Is there a reason you can't use `IP_REGEX` to validate input here?

Reply:
I mostly just copied over from old code

Reply:
The other places checking if it includes a `.` should probably be updated to use IP_REGEX.

Reply:
Yeah Ill update it to use that in a sec, first im moving IP_REGEX to IPTools